analysis_driver/reader/demultiplexing_parsers.py

```diff
@@ -93,9 +103,78 @@ def parse_seqtk_fqchk_file(fqchk_file, q_threshold):
         lo_q = 0
         hi_q = 0
         for i, h in enumerate(header[9:]):
-            # header are %Q2
+            #header are %Q2
             if int(h[2:]) < q_threshold:
                 lo_q += int(all_cycles[9+i])
             else:
                 hi_q += int(all_cycles[9+i])
-        return nb_read, nb_base, lo_q, hi_q
+        return  nb_read, nb_base, lo_q, hi_q
+
+
+def get_focal_species(sample_id):
+    myFocalSpecies = get_species_from_sample(sample_id)
+    return myFocalSpecies
+
+def parse_fastqscreen_file(filename, myFocalSpecies):
+    """
+    parse the fastq screen outfile
+    :return dict: the maximum number of reads mapped uniquely (singly or multiple times) to a contaminant species
+    :return float: % reads unmapped to focal Species
+    :return float: % reads with no hits to any of the genomes provided
+    :return int: number of reads mapped in total
+    """
+    file = open(filename)
+    lines = file.readlines()
+
+    contaminantsUniquelyMapped = {}
+    focalSpeciesPercentUnmapped = ''
+    total_reads_mapped = int(((lines[0]).split(': ')[2]).rstrip('\n'))
+    Hit_no_genomes = float((lines[-1]).split(': ')[1])
+    speciesResults = (lines[2:-2])
+    speciesList = []
+    for result in speciesResults:
+        speciesName = result.split('\t')[0]
+        speciesName = speciesName.replace('_',' ')
+        speciesList.append(speciesName)
+
+    if myFocalSpecies in speciesList:
+        for result in speciesResults:
+            speciesName = result.split('\t')[0]
+            speciesName = speciesName.replace('_',' ')
+            speciesResults = result.split('\t')[1:12]
+
+            if speciesName != myFocalSpecies:
+                numberUniquelyMapped = int(result.split('\t')[4]) + int(result.split('\t')[6])
+                contaminantsUniquelyMapped[speciesName] = numberUniquelyMapped
+            elif speciesName == myFocalSpecies:
+                focalSpeciesPercentUnmapped = float(speciesResults[2])
+        contaminantsUniquelyMapped = {k:v for k,v in contaminantsUniquelyMapped.items() if v != 0}
+        ELEMENT_SPECIES_CONTAMINATION = {ELEMENT_CONTAMINANT_UNIQUE_MAP:contaminantsUniquelyMapped,
```

Comment:
I don't think this is how `ELEMENT_SPECIES_CONTAMINATION` was meant to be used.
Uppercase variables are meant to be constant and here this one is reassigned with a dict